components/hab/src/command/origin/transfer.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{api_client::{self,
+                         Client},
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+use reqwest::StatusCode;
+
+pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str, account: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Transferring,
+              format!("ownership of origin {} to {}.", origin, account))?;
+
+    match api_client.transfer_origin_ownership(origin, token, account) {
+        Ok(_) => {
+            ui.status(Status::Transferred, "ownership successfully!".to_string())
+              .or(Ok(()))
+        }
+        Err(api_client::Error::APIError(StatusCode::FORBIDDEN, msg)) => {
+            ui.fatal("Failed to transfer origin ownership!")?;
+            // TODO: remove the "the operation has already been completed" phrase
+            // when this issue is closed: https://github.com/habitat-sh/builder/issues/1273
+            ui.fatal("Either you are not the current owner, the operation requested has already \
+                      been completed, or the new owner is not yet a member.")?;
+            Err(Error::APIClient(api_client::Error::APIError(StatusCode::FORBIDDEN, msg)))
+        }
+        Err(api_client::Error::APIError(StatusCode::UNAUTHORIZED, msg)) => {
+            ui.fatal("Failed to transfer origin ownership!")?;
+            Err(Error::APIClient(api_client::Error::APIError(StatusCode::UNAUTHORIZED, msg)))
+        }
```

Comment:
Minor - Could we use [bindings](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#-bindings) to simplify this code a bit?

Something like
```rust
        Err(err @ api_client::Error::APIError(StatusCode::UNAUTHORIZED, _)) => {
            ui.fatal("Failed to transfer origin ownership!")?;
            Err(Error::APIClient(err))
        }
```
This would apply to the other match arms as well.

Reply:
That is such a cool feature @davidMcneil - thanks for pointing it out!! I refactored in https://github.com/habitat-sh/habitat/pull/7331/commits/a433ab1a0ae9edde576da0a033f958cc6b6a02ed